simhash/__init__.py

```diff
@@ -8,44 +8,59 @@
 import numbers
 import collections
 from itertools import groupby
+import numpy as np
 
 if sys.version_info[0] >= 3:
     basestring = str
     unicode = str
     long = int
+
+    def int_to_bytes(n, length):
+        return n.to_bytes(length, 'big')
+
+    def bytes_to_int(b):
+        return int.from_bytes(b, 'big')
 else:
     range = xrange
 
+    def int_to_bytes(n, length):
+        return '{:0{}x}'.format(n, length * 2).decode('hex')
+
+    def bytes_to_int(b):
+        return int(b.encode('hex'), 16)
 
 def _hashfunc(x):
-    return int(hashlib.md5(x).hexdigest(), 16)
+    return hashlib.md5(x).digest()
 
 
 class Simhash(object):
+    # Constants used in calculating simhash. Larger values will use more RAM.
+    large_weight_cutoff = 50
```

Comment:
Could you elaborate on how 50 is calculated? Also if can you help add unit tests for both cases (small weight and large weight), that would be perfect.

Reply:
I selected both of those values based on where they hit diminishing returns on my machine based on benchmarks. The total RAM needed in the worst case is roughly `batch_size * large_weight_cutoff * f_bytes` (equals about 80KB), so I didn't want those numbers to be much larger than they need to be -- but I also suspect that anything in the right ballpark is totally fine.

batch_size I think wants to be in the 100 or 200 range -- we want to amortize the costs of building the numpy array, but once you're dividing the per-batch cost by a hundred or so there's not much left to win from larger batches. 200 was a little better than 100 on my machine, after that any gain was lost in statistical noise.

large_weight_cutoff is a tradeoff -- features below a certain size will be faster with the `batch.append(h * w)` branch, and above a certain size they'll be faster with the `sums.append(self._bitarray_from_bytes(h) * w)` branch. On my machine it looks like anywhere between like 30 and 300 was a tossup, so I went with 50 to err on the side of using less RAM.

I think these are basically fine values and values plus or minus 2x would also be fine. 🤷‍♂️

> Also if you can help add unit tests for both cases (small weight and large weight), that would be perfect.

I pushed a new version that adds a test for a large feature list with small weights, and large feature list with large weights.